### PR TITLE
feat: add Airweave tool integration with advanced search features

### LINKED ISSUE #20110: [Feature Request]: Add Airweave Integration as a Tool
### Feature Description

Add Airweave tool integration to enable agents to search across data synced from 30+ sources.

Airweave is an open-source platform that automatically syncs data from multiple sources (Google Drive, Notion, GitHub, databases, etc.) and provides unified search with advanced retrieval.

### Reason

Don't think anyone has ever taken then time to build it, we will open a PR with working integration.

### Value of Feature

- Instant access to 30+ pre-built data source connectors for agent builders
- Automatic syncing keeps data up-to-date without manual pipelines
- Advanced search (hybrid retrieval, temporal relevance, reranking)
- Built-in RAG answer generation
- Open-source and self-hostable

Perfect for enterprise knowledge bases where data needs to stay continuously synced.

### PR DESCRIPTION
**Pull request by EwanTauran**
## Description

This PR adds Airweave as a tool integration, enabling LlamaIndex agents to search across data automatically synced from 30+ sources.

**What is Airweave?**
Airweave is an open-source platform that syncs data from multiple sources (Google Drive, Notion, GitHub, databases, APIs, etc.) and provides unified search with advanced retrieval capabilities.

**Implementation:**
- **AirweaveToolSpec** with 5 tool functions:
  - `search_collection`: Simple search with default settings
  - `advanced_search_collection`: Full control over retrieval parameters
  - `search_and_generate_answer`: RAG-style direct answers
  - `list_collections`: Discover available collections
  - `get_collection_info`: Get collection details

- **Advanced search features:**
  - Multiple retrieval strategies (hybrid, neural, keyword)
  - Temporal relevance weighting for recent content
  - Query expansion for better recall
  - Auto-interpret filters from natural language
  - LLM-based reranking for improved relevance
  - Natural language answer generation

**Testing & Documentation:**
- 13 passing unit tests with comprehensive coverage
- Full README with usage examples
- Jupyter notebook demonstrating all features
- Production tested with real data

**Links:**
- Airweave: https://airweave.ai/
- Docs: https://docs.airweave.ai/
- GitHub: https://github.com/airweave-ai/airweave

Fixes #20110

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes - Set to version 0.1.0 in pyproject.toml
- [ ] No

## Type of Change

Please delete options that are not relevant.

[x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- [x] I added new unit tests to cover this change

**Testing details:**
```bash
cd llama-index-integrations/tools/llama-index-tools-airweave
uv run pytest tests/ -v
# Result: 13 passed, 0 failed
```

All tests use mocked Airweave SDK calls and cover:
- Class inheritance and initialization
- All 5 tool functions (search, advanced search, RAG answers, list, get info)
- Edge cases (empty results, missing completions)
- Both dict and object response parsing

Additionally tested with production Airweave instance and real data.

## Suggested Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `uv run make format; uv run make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        Initialize with Airweave API credentials.
+
+        Args:
+            api_key: Your Airweave API key from the dashboard
+            base_url: Optional custom base URL for self-hosted instances
+            framework_name: Framework name for analytics (default: "llamaindex")
+            framework_version: Framework version for analytics
+
+        """
+        from airweave import AirweaveSDK
```
Comment (AstraBert) [inline llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: Can we place imports at the top? :)

File: llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                          (e.g., 'finance-data-ab123')
+            query: The search query in natural language
+            limit: Maximum number of results to return (default: 10)
+            offset: Number of results to skip for pagination (default: 0)
+
+        Returns:
+            List of Document objects containing search results with metadata
+
+        """
+        from airweave import SearchRequest
```
Comment (AstraBert) [inline llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: imports at the top :)

File: llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+                generate_answer=True,
+                rerank=use_reranking,
+            ),
+        )
+
+        if hasattr(response, "completion") and response.completion:
+            return response.completion
+        else:
+            # Fallback if no answer generated
+            return "No answer could be generated from the search results."
```
Comment (AstraBert) [inline llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: raise an error? or a warning maybe? I would personally be in favor of doing:

```suggestion
        else:
            # Fallback if no answer generated
            warnings.warn("No answer could be generated from the search results", UserWarning)
            return None
```

And then type the return type of the function as: `Optional[str]` so that the user knows, when the result is None, that no answer was generated

Comment (AstraBert) [reply llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: (this means importing also `warnings` at the top)

Comment (EwanTauran) [reply llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: done!

File: llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py (hunk 2 above)
Comment (EwanTauran) [reply llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: done!

File: llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py (hunk 1 above)
Comment (EwanTauran) [reply llama-index-integrations/tools/llama-index-tools-airweave/llama_index/tools/airweave/base.py]: done.

**Review by AstraBert — APPROVED** [review]
